mithril-core/src/merkle_tree.rs

```diff
@@ -377,12 +667,48 @@ mod tests {
         }
 
         #[test]
-        fn test_bytes_tree_commitment((t, values) in arb_tree(5)) {
-            let encoded = bincode::serialize(&t.to_commitment()).unwrap();
-            let decoded: MerkleTreeCommitment::<Blake2b<U32>> = bincode::deserialize(&encoded).unwrap();
-            let tree_commitment = MerkleTree::<Blake2b<U32>>::create(&values).to_commitment();
+        fn test_create_batch_proof((t, values) in arb_tree(30)) {
+            let mut mt_index_list :Vec<usize> = Vec::new();
+            for (i, _v) in values.iter().enumerate() {
+                let ind = Some(i as usize);
+                mt_index_list.push(ind.unwrap());
```

Comment:
Again, what is the purpose of doing `Some()` and the `unwrap()`?
```suggestion
                mt_index_list.push(i);
```